### PR TITLE
chore: add vesting

### PR DESCRIPTION
**Pull request by GAtom22**

### CODE REVIEW

File: precompiles/stateful/Vesting.sol [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+struct Period {
+    int64 length;
+    Coin[] amount;
+}
+
+/// @author Evmos Team
+/// @title Vesting Precompiled Contract
+/// @dev The interface through which solidity contracts will interact with vesting.
+/// We follow this same interface including four-byte function selectors, in the precompile that
+/// wraps the pallet.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:30]: What are we referring to when we say pallet? Is it something that references Moonbeam precompiles?

File: precompiles/stateful/Vesting.sol [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+/// @custom:address 0x0000000000000000000000000000000000000803
+interface VestingI {
+    /// @dev Defines a method for creating a new clawback vesting account.
+    /// @param funderAddress The address of the account that will fund the vesting account.
+    /// @param vestingAddress The address of the account that will receive the vesting account.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:35]:
```suggestion
    /// @param vestingAddress The address of the account that will be converted into a vesting account.
```

File: precompiles/stateful/Vesting.sol [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    /// QUERIES
+
+    /// @dev Defines a query for getting the balances of a vesting account.
+    /// @param vestingAddress The address of the vesting account.
+    function balances(
+        address vestingAddress
+    ) external view returns (Coin[] memory locked, Coin[] memory unvested, Coin[] memory vested);
+
+    /// @dev Defines an event that is emitted when a clawback vesting account is created.
+    /// @param funderAddress The address of the account that funded the vesting account.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:92]:
```suggestion
    /// @param funderAddress The address of the account that can fund the vesting account.
```

File: precompiles/stateful/Vesting.sol [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    /// @param vestingAddress The address of the account that received the vesting account.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:93]:
```suggestion
    /// @param vestingAddress The address of the account that has been converted to a vesting account.
```

File: precompiles/stateful/Vesting.sol [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    event CreateClawbackVestingAccount(
+        address indexed funderAddress,
+        address indexed vestingAddress
+    );
+
+    /// @dev Defines an event that is emitted when a clawback vesting account is funded.
+    /// @param funderAddress The address of the account that funded the vesting account.
+    /// @param vestingAddress The address of the account that received the vesting account.
+    /// @param startTime The time at which the vesting account will start.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:102]:
```suggestion
    /// @param startTime The time at which the vesting will start.
```

File: precompiles/stateful/Vesting.sol [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    /// @param accountAddress The address of the vesting account.
+    /// @param destAddress The address of the account that received the clawed back coins.
+    event Clawback(
+        address indexed funderAddress,
+        address indexed accountAddress,
+        address destAddress
+    );
+
+    /// @dev Defines an event that is emitted when a vesting account's funder is updated.
+    /// @param funderAddress The address of the account that funded the vesting account.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:124]:
```suggestion
    /// @param funderAddress The address of the account that can fund the vesting account.
```

**Review by 0xstepit — COMMENTED** [review]
Nice job @GAtom22, just minor comments on the docstring

**Review by Vvaradinov — APPROVED** [review]

**Pull request merged**